### ISSUE TITLE
Import Collapse includes Popper.js and should not

### ISSUE THREAD

**Issue by tkrotoff** [reactstrap/reactstrap#1246]
- components: `Collapse`
- reactstrap version `#6.4.0`
- import method `es`
- react version `#16.5.2`
- bootstrap version `#4.1.1`
- webpack version `#4.20.2`

### What is happening?

```JS
import { Collapse } from 'reactstrap';
```

makes my app includes Popper.js while it should not because [Collapse.js](https://github.com/reactstrap/reactstrap/blob/master/src/Collapse.js) does not use it (and thanks to [webpack tree shaking](https://github.com/reactstrap/reactstrap/blob/6.4.0/package.json#L12)?)

![screen shot 2018-10-04 at 00 39 04](https://user-images.githubusercontent.com/643434/46443455-f58fba00-c76d-11e8-883d-a42e4541874c.png)

This most probably happens for other components that do not depend on Popper.js

### What should be happening?

If I use Collapse from React-Bootstrap (very similar  [Collapse.js](https://github.com/react-bootstrap/react-bootstrap/blob/v1.0.0-beta.1/src/Collapse.js) implementation):

```JS
import { Collapse } from 'react-bootstrap';
```

Popper.js is not included in my bundle.js (and obviously the collapse works). My bundle.js drops from 446 KB to 343 KB (size on file system) => 30% increase.

### Steps to reproduce issue

Create a simple React app and use this Collapse facade:

```TypeScript
import React from 'react';
//import { Collapse as ReactBootstrapCollapse } from 'react-bootstrap';
import { Collapse as ReactstrapCollapse } from 'reactstrap';

interface Props {
  in: boolean;
}

/*
const Collapse: React.SFC<Props> = props => {
  return (
    <ReactBootstrapCollapse in={props.in}>
      {props.children}
    </ReactBootstrapCollapse>
  );
};
*/

const Collapse: React.SFC<Props> = props => {
  return (
    <ReactstrapCollapse isOpen={props.in}>
      {props.children}
    </ReactstrapCollapse>
  );
};

export default Collapse;
```

then use webpack-bundle-analyzer: `webpack -p --profile --json > stats.json && webpack-bundle-analyzer stats.json build` or check the resulting bundle.js on the file system.

FYI [React-Bootstrap uses](https://github.com/react-bootstrap/react-bootstrap/blob/v1.0.0-beta.1/package.json#L123) [@react-bootstrap/react-popper](https://www.npmjs.com/package/@react-bootstrap/react-popper) which seems to be a "special" version of react-popper, don't know if it's related.

**Comment by Sly321** [comment]
Since you import the Collapse with
`import { Collapse } from "reactstrap"`, the Code will go for the index.js in reactstrap, where all the imports are found. And all the imports are loaded. It is like this import Syntax: 
```javascript
import reactstrap from "reactstrap"
reactstrap.Collapse // dostuff
```
If you import it like `import Collapse from "reactstrap/src/Collapse"`, you only import the Collapse and it's dependencies. Which is not popper.js.

I know "tree shaking should eliminate the dead code parts". I actually don't really know what this tree shake is going todo on such index.js exports. Generally it's a better solution to import only that what you need.

**Comment by tkrotoff** [comment]
>  I actually don't really know what this tree shake is going todo on such index.js exports

It shouldn't be an issue, see index.js from libraries that are known to work with webpack tree shaking:
- https://github.com/lodash/lodash/blob/4.17.11-es/lodash.js
- https://github.com/date-fns/date-fns/blob/v2.0.0-alpha.21/src/index.js
- https://github.com/react-bootstrap/react-bootstrap/blob/v1.0.0-beta.1/src/index.js

> it's a better solution to import only that what you need.

Talking about `import Button from 'reactstrap/lib/Button`? That's not the sense of history and where JS is going. You expose your internal code structure (`lib/Button`) to the user.

Anyway, I think I've found a lead: 
https://github.com/reactstrap/reactstrap/blob/0dc2acda96913db56418a880b9ebd807a9c05076/package.json#L6

reactstrap generates a unique file [`dist/reactstrap.es.js`](https://unpkg.com/reactstrap@6.5.0/dist/reactstrap.es.js). From my understanding, you should not "generate" a one dist esnext file.

https://webpack.js.org/guides/tree-shaking/:
> [Tree shaking] relies on the static structure of ES2015 module syntax, i.e. import and export

Inside `dist/reactstrap.es.js` you don't have export/import anymore: everything is flatten: not possible to tree shake.

See how others do it:
- https://github.com/lodash/lodash/blob/4.17.11-es/package.json#L10
- https://github.com/date-fns/date-fns/blob/v2.0.0-alpha.21/package.json#L18 and https://unpkg.com/date-fns@2.0.0-alpha.21/esm/index.js
- https://github.com/react-bootstrap/react-bootstrap/blob/v1.0.0-beta.1/package.json#L12 and https://unpkg.com/react-bootstrap@1.0.0-beta.1/es/index.js

Instead it should be:
```JSON
"jsnext:main": "src/index.js",
```

**Comment by tkrotoff** [comment]
btw,
https://github.com/reactstrap/reactstrap/blob/6.5.0/src/index.js

```JS
import Button from './Button';
...
export {
  Button,
  ...
};
```

I think it can be simply rewritten:
```JS
export Button from './Button'
```
and divide by 2 the index.js number of lines.
(no impact on webpack tree shaking, just syntax sweetness)

**Comment by Sly321** [comment]
I didn't saw it that way, but i guess you're right! I learned something, thanks. 👍

**Comment by TheSharpieOne** [comment]
This should be fixed in `7.0.2`. I updated rollup and added `@babel/plugin-proposal-export-default-from` and `@babel/plugin-proposal-export-namespace-from`.
I changed the imports to export so it doesn't export a single large object (which apparently was the issue).
